src/core/tree_methods.ts

```diff
@@ -0,0 +1,85 @@
+import * as _ from 'lodash'
+
+export class TreeNode<T, B, A> {
```

Comment:
Almost there :D
You just need one generic type and that woule be T. There is no reason for A nor B, unless I'm missing something. You have the contents (which I would even be inclined to call bluntly "payload") of type T and children of type TreeNode<T>. Is_done is part of the payload.

Also: we don't set is_done, we swap the whole node.

There is a question if we should allow a Tree<X> to contain Tree<Y>. I'd say that nope, until proven otherwise.

Sidenote: does the code even compile in the current form?

Reply:
ok so I removed setIs_done test and the method,
changed the type to just T and
removed is_done field, as it's going to be passed as part of 'content'

Reply:
it doesnt compile :(
"""
Generic type 'TreeNode<T>' requires 1 type argument(s).  TS2314

     5 |   children: T;
     6 | 
  >  7 |   constructor(content:string, children: TreeNode[] = []){
       |                                         ^
     8 |     this.content = content;
"""

Reply:
I made it compile with a new constructor. Now the methods dont compile as they are not generic.

Reply:
Yup, we need generic methods and `TreeNodes<T>` as child arguments

Reply:
You can even write `Array<TreeNode<T>>` :D